src/typescript/09-interfaces/interfaces.md

```diff
@@ -97,76 +216,49 @@ Hint: the interface should have properties and types:
   - ``state``(string)
   - ``zip`` (string)
 
-### The solution
+### Solution
 
 <details>
 <summary>Click to see the solution</summary>
 
-✏️ Update `5a-dino-park.ts` to the following:
+✏️ Update `dino-park.ts` to the following:
 
-```typescript
-import Address from "./5a-address";
+@sourceref ../../../exercises/typescript/09-interfaces/01-solution/src/dino-park.ts
+@highlight 4-6
 
-interface DinoPark {
-  name: string;
-  image?: string;
-  address: Address;
- }
-
-export default DinoPark;
-```
-
-@highlight 3-7
-
-✏️ Update `5a-address.ts` to the following:
-
-```typescript
-interface Address {
-  street: string;
-  city: string;
-  state: string;
-  zip: string
-}
-
-export default Address;
-```
+✏️ Update `address.ts` to the following:
 
+@sourceref ../../../exercises/typescript/09-interfaces/01-solution/src/address.ts
 @highlight 2-5
 
 </details>
 
 Make sure you have this solution implemented correctly before moving on to the next exercise.
 
-## Exercise 2: Write a Function With a Parameter Described by Interface
-
-### The Problem
+### Exercise 2: Write a Function With a Parameter Described by Interface
```

Comment:
It looks like this exercise only covers what’s under “Interfaces in functions”, so the exercise should come after that section.

So overall, I would consider having three objectives on this page:

1. Describe objects with interfaces [the first 3 concepts + exercise 1]
2. Describe functions and their parameters [next 2 concepts + exercise 2]
3. Assert types on objects [last concept, would be nice to add a exercise but ok to leave for now]

_Or_ I would move the “Type assertion” section to be under the first objective, not as its own. I think that could be a good way to put all the object stuff in the first objective and all the function/parameter stuff in the second objective.